subsys/net/ip/utils.c

```diff
@@ -313,7 +313,10 @@ int net_addr_pton(sa_family_t family, const char *src,
 					return -EINVAL;
 				}
 
-				src++;
+				if (src) {
```

Comment:
This doesn't look like a code written by human to me. An automated program rewrite tool could do that. A human would refactor to:

~~~
if (src) {
  src++
} else {
  if (i < expected_groups - 1) ...
}
~~~

P.S. Rule of thumb "try to change as little as possible lines/chars in a patch" has its limits too, sometimes it makes sense to do more changes than the absolute minimum necessary, to maintain code quality/clarity.

Reply:
> } else { if (i < expected_groups - 1) ...

I could not really figure out from where did you got the "else" branch.
With this new code in this commit, we bail out after the continue statement.

Reply:
I got it from looking at a bit more context than just a line Coverity complains about (apparently).

The code you want to land is:

~~~
if (!src && FOO) {
}
if (src) {
}
~~~

Again, humans don't normally write such code. Why instruct compiler to test the same condition twice, of instead if-else can be used?

Reply:
Ah, got it. Sure, that can be refactored.